flagsmith/resource_segment.go

```diff
@@ -0,0 +1,269 @@
+			Required:            true,
+			MarkdownDescription: "Type of the rule",
+			Type:                types.StringType,
+		},
+		"conditions": {
+			Optional:            true,
+			MarkdownDescription: "List of conditions for the nested rule",
+			Attributes:          conditions,
+		},
+	})
+
+	return tfsdk.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Flagsmith Segment",
+
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Computed:            true,
+				MarkdownDescription: "ID of the segment",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+				Type: types.Int64Type,
+			},
+			"uuid": {
+				Computed:            true,
+				MarkdownDescription: "UUID of the segment",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+				Type: types.StringType,
+			},
+			"project_id": {
+				Computed:            true,
+				MarkdownDescription: "ID of the project",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+				Type: types.Int64Type,
+			},
+			"feature_id": {
+				Computed:            true,
+				Optional:            true,
+				MarkdownDescription: "Set this to create a feature specific segment",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+				Type: types.Int64Type,
+			},
+			"name": {
+				Required:            true,
+				MarkdownDescription: "Name of the segment",
+				Type:                types.StringType,
+			},
+			"description": {
+				Optional:            true,
+				MarkdownDescription: "Description of the segment",
+				Type:                types.StringType,
+			},
+			"project_uuid": {
+				MarkdownDescription: "UUID of project the segment belongs to",
+				Required:            true,
+				Type:                types.StringType,
+			},
+			"rules": {
+				MarkdownDescription: "Rules for the segment",
+				Required:            true,
+				Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
+					"type": {
+						Required:            true,
+						MarkdownDescription: "Type of the rule, can be of: `ALL`, `ANY`, `NONE`",
+						Type:                types.StringType,
+					},
+					"rules": {
+						Optional:            true,
+						MarkdownDescription: "List of Nested Rules",
+						Attributes:          nestedRules,
+					},
+
+					"conditions": {
+						Optional:            true,
+						MarkdownDescription: "Conditions for the rule",
+						Attributes:          conditions,
+					},
+				}),
+			},
+		},
+	}, nil
+
+}
+
+
+
+func (r *segmentResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data SegmentResourceData
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clientSegment := data.ToClientSegment()
+
+	// Create the segment
```

Comment:
Minor point: I think this comment is unnecessary, the method we're calling on the line below is literally called `CreateSegment` - that is clear enough, right? Unless I'm missing something?

Reply:
Yeah, true